packages/types/src/codec/Enum.ts

```diff
@@ -206,7 +242,7 @@ export class Enum implements IEnum {
 
         for (let i = 0; i < keys.length; i++) {
           const key = keys[i];
-          const name = stringUpperFirst(stringCamelCase(key.replace(' ', '_')));
+          const name = memoizedPropertySuffix(key);
```

Comment:
I view this in the same way as the one from `Set`, so will revert this after merge.

(There is probably something to be said for a generalized solution with Struct, Set, Enum, ... where they all follow the same pattern, but as it stands it is micro in the lesser-used paths)

Reply:
I found that this was an interesting case where it did make sense. `stringCamelCase` does a few regular expressions internally, but always act on the same inputs. It was showing as a significant contributor of CPU usage in the profiler before this change.

Reply:
If it calls it on the same enum, doing the same keys - there is another issue, i.e. the like-definitions are re-created instead of re-used. That should be solved. At this layer, for a specific definition, it _should_ only be created once. In that case the cache hits here should be close to 0.

(It is quite possible that that assumption doesn't hold anymore after the metadata v14 conversion, but that is still a bug/unintended consequence then.)

Reply:
Ok, this is not something new, but certainly something that should be done - so yes, indeed base aalsses are cached, e.g. `NodeRuntimeSessionKeys`. 

However what is not cached and re-created are stuff that doesn't appear in the lookup registry explicitly, e.g. `Vec<NodeRuntimeSessionKeys>` will be recreated each time it is used. (Same with embedded enums, etc.) So there is an opportunity to improve on a per-registry basis. The biggest overhead is indeed the dance without going from definition -> Class while parsing.